src/Microsoft.Azure.SignalR.Common/Utilities/ConnectionStringParser.cs

```diff
@@ -180,5 +215,35 @@ private static AccessKey BuildAccessKey(Uri uri, Dictionary<string, string> dict
             }
             throw new ArgumentException(MissingAccessKeyProperty, AccessKeyProperty);
         }
+
+        private static AccessKey BuildAzureAdAccessKey(Uri uri, Dictionary<string, string> dict)
+        {
+            if (dict.TryGetValue(ClientIdProperty, out var clientId))
```

Comment:
extract common code with BuildAadAccessKey?

Reply:
Not a good idea since they're legacy codes.